src/PhpParser/Node/NodeFactory.php

```diff
@@ -291,6 +294,11 @@ public function createPromotedPropertyParam(PropertyMetadata $propertyMetadata):
         $propertyFlags = $propertyMetadata->getFlags();
         $param->flags = $propertyFlags !== 0 ? $propertyFlags : Modifiers::PRIVATE;
 
+        // make readonly by default
+        if ($this->phpVersionProvider->isAtLeastPhpVersion(PhpVersionFeature::READONLY_PROPERTY)) {
+            $param->flags |= Modifiers::READONLY;
```

Comment:
I prefer to have a flag to avoid BC break so we need to choose to apply readonly or not, eg: `$markReadonly = false`

```php
public function createPromotedPropertyParam(
       PropertyMetadata $propertyMetadata, 
       bool $markReadonly = false
): Param
```

with default `true` or `false` depends on what you concern, but give user ability to choose.

Reply:
This is not ment to be configured, as default behavior.